File: stageTemplates/src/app/models/grandeur.ts
import { TypeGrandeur } from "./typegrandeur";

export class Grandeur {
    
    id: number;
    nomGrandeur?: string | null;
    nomAbrege?: string | null;
    nomComplet?: string | null;
    typeGrandeurId: number;
    typeGrandeur?: TypeGrandeur | null;
    unite?: string | null;
    localisationBarr?: string | null;
    modeAcquisition?: string | null;
    frequenceMesure?: string | null;
    precisionMesure?: number | null;
    coordonneeX?: number | null;
    coordonneeY?: number | null;
    coordonneeZ?: number | null;
    minimum?: number | null;
    maximum?: number | null;
    valeurDexclusion?: number | null;
    processorId?: number | null;
    processorUnitId?: number | null;
    processorName?: string | null;
    numberDecimalDigits?: number | null;
    codeAppareil?: number | null;
    heureMBI?: number | null;
    dateMBI?: number | null;
    valeurMBI?: number | null;
    piezomPlusProche?: string | null;
    lectureDeBase?: number | null;
    longueur?: number | null;
    ancre?: number | null;
    calibre?: number | null;
    klCalibrage?: number | null;
    lineaireZero?: number | null;
    siteId?: number | null;
    site?: any; 
    
    constructor(
        id: number = 0,
        nomGrandeur: string | null = null,
        nomAbrege: string | null = null,
        nomComplet: string | null = null,
        typeGrandeurId: number = 0,
        localisationBarr: string | null = null,
        modeAcquisition: string | null = null,
        frequenceMesure: string | null = null,
        precisionMesure: number | null = null,
        coordonneeX: number | null = null,
        coordonneeY: number | null = null,
        coordonneeZ: number | null = null,
        minimum: number | null = null,
        maximum: number | null = null,
        valeurDexclusion: number | null = null,
        processorId: number | null = null,
        processorUnitId: number | null = null,
        processorName: string | null = null,
        numberDecimalDigits: number | null = null,
        codeAppareil: number | null = null,
        heureMBI: number | null = null,
        dateMBI: number | null = null,
        valeurMBI: number | null = null,
        piezomPlusProche: string | null = null,
        lectureDeBase: number | null = null,
        longueur: number | null = null,
        ancre: number | null = null,
        calibre: number | null = null,
        klCalibrage: number | null = null,
        lineaireZero: number | null = null,
        siteId: number | null = null
      ) {
        this.id = id;
        this.nomGrandeur = nomGrandeur;
        this.nomAbrege = nomAbrege;
        this.nomComplet = nomComplet;
        this.typeGrandeurId = typeGrandeurId;
        this.localisationBarr = localisationBarr;
        this.modeAcquisition = modeAcquisition;
        this.frequenceMesure = frequenceMesure;
        this.precisionMesure = precisionMesure;
        this.coordonneeX = coordonneeX;
        this.coordonneeY = coordonneeY;
        this.coordonneeZ = coordonneeZ;
        this.minimum = minimum;
        this.maximum = maximum;
        this.valeurDexclusion = valeurDexclusion;
        this.processorId = processorId;
        this.processorUnitId = processorUnitId;
        this.processorName = processorName;
        this.numberDecimalDigits = numberDecimalDigits;
        this.codeAppareil = codeAppareil;
        this.heureMBI = heureMBI;
        this.dateMBI = dateMBI;
        this.valeurMBI = valeurMBI;
        this.piezomPlusProche = piezomPlusProche;
        this.lectureDeBase = lectureDeBase;
        this.longueur = longueur;
        this.ancre = ancre;
        this.calibre = calibre;
        this.klCalibrage = klCalibrage;
        this.lineaireZero = lineaireZero;
        this.siteId = siteId;
      }
  }
  

File: CondorV/CondorV/Models/BD/Grandeur.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace CondorV.Models.BD
{
    public class Grandeur
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }
        public string? NomGrandeur { get; set; }
        public string? NomAbrege { get; set; }
        public string? NomComplet { get; set; }

        public long TypeGrandeurId { get; set; }
        public TypeGrandeur? TypeGrandeur {get;set;}
        public string? Unite { get; set; }
        public int? LocalisationBarrId { get; set; }
        //[JsonIgnore]
        public LocalisationBarr? LocalisationBarr { get; set; }
        public string? ModeAcquisition { get; set; }
        public string? FrequenceMesure { get; set; }
        public int? PrecisionMesure { get; set; }
        public double? CoordonneeX { get; set; }
        public double? CoordonneeY { get; set; }
        public double? CoordonneeZ { get; set; }
        public int? Minimum { get; set; }
        public int? Maximum { get; set; }
        public int? ValeurDexclusion { get; set; }
        public double? ProcessorId { get; set; }
        public double? ProcessorUnitId { get; set; }
        public string? ProcessorName { get; set; }
        public int? NumberDecimalDigits { get; set; }
        public double? CodeAppareil { get; set; }
        public float? HeureMBI { get; set; }
        public float? DateMBI { get; set; }
        public float? ValeurMBI { get; set; }
        public string? PiezomPlusProche { get; set; }
        public long? LectureDeBase { get; set; }
        public double? Longueur { get; set; }
        public double? Ancre { get; set; }
        public double? Calibre { get; set; }
        public double? KlCalibrage { get; set; }
        public float? LineaireZero { get; set; }
        public long? SiteId { get; set; }
        [JsonIgnore]
        public ICollection<Mesure>? Mesures { get; set; }
        //[JsonIgnore]
        public Site? Site { get; set; }
    }
}
